app/qbftdebug.go

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package app
+
+import (
+	"bytes"
+	"compress/gzip"
+	"net/http"
+	"sync"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/log"
+	pbv1 "github.com/obolnetwork/charon/core/corepb/v1"
+)
+
+const maxQBFTDebugger = 2 * (1 << 20) // 2 MB
+
+// qbftDebugger buffers up to 2MB worth of sniffed qbft messages in a fifo buffer serving them as a gzipped
+// *pbv1.SniffedConsensusSets protobuf on request.
+type qbftDebugger struct {
+	mu        sync.Mutex
+	totalSize int
+	sets      []*pbv1.SniffedConsensusMsgs
```

Comment:
why not using `pbv1.SniffedConsensusSets` directly here?

Reply:
because then one needs to initialise it, slices work without initialisation